pkg/cmd/create_job.go

```diff
@@ -0,0 +1,238 @@
+	PreviewFormat string
+	RawPreview    bool
+}
+
+// NewCreateJobOptions provides an instance of CreateJobOptions with default values.
+func NewCreateJobOptions(streams genericclioptions.IOStreams) *CreateJobOptions {
+	return &CreateJobOptions{
+		configFlags:       genericclioptions.NewConfigFlags(true),
+		previewPrintFlags: genericclioptions.NewJSONYamlPrintFlags(),
+		printFlags:        genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:         streams,
+	}
+}
+
+// AddFlags adds a flag to the flag set.
+func (o *CreateJobOptions) AddFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.from, "from", o.from, "The name of the resource to create a Job from (only cronjob is supported).")
+
+	// original flags
+	flags.BoolVarP(&o.Preview, "preview", "P", false,
+		"If true, display the object YAML|JSON by preview window for fuzzy finder selector.")
+	flags.StringVar(&o.PreviewFormat, "preview-format", "yaml",
+		"Preview window output format. One of json|yaml.")
+	flags.BoolVar(&o.RawPreview, "raw-preview", false,
+		"If true, display the unsimplified object in the preview window. (default is simplified)")
+}
+
+// Complete sets all information required for get logs.
+func (o *CreateJobOptions) Complete(cmd *cobra.Command, args []string) error {
+	client, err := kubernetes.NewClient(o.configFlags)
+	if err != nil {
+		return fmt.Errorf("failed to new Kubernetes client: %w", err)
+	}
+
+	if o.from == "" {
+		return errors.New("--from option is required, only supported job from cronjob")
+	}
+
+	if o.from != "cronjob" {
+		return errors.New("must specify resource, only supported cronjob")
+	}
+
+	if len(args) >= 1 {
+		o.name = args[0]
+	}
+
+	o.jobClient = client.BatchV1()
+	o.Builder = resource.NewBuilder(o.configFlags)
+
+	kubeConfig := o.configFlags.ToRawKubeConfigLoader()
+
+	namespace, _, err := kubeConfig.Namespace()
+	if err != nil {
+		return fmt.Errorf("faild to get namespace from kube config: %w", err)
+	}
+
+	o.namespace = namespace
+
+	printer, err := o.printFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.printObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
+	}
+
+	return nil
+}
+
+// Validate ensures that all required arguments and flag values are provided.
+func (o *CreateJobOptions) Validate() error {
+	return nil
+}
+
+// Run execute fizzy finder and create job from cronJob.
+func (o *CreateJobOptions) Run(ctx context.Context) error {
+	infos, err := o.Builder.
+		Unstructured().
+		NamespaceParam(o.namespace).DefaultNamespace().
+		ResourceTypes(o.from).
+		SelectAllParam(true).
+		Flatten().
+		Latest().
+		Do().
+		Infos()
+	if err != nil {
+		return fmt.Errorf("failed to list cronJobs: %w", err)
+	}
+
+	var printer printers.ResourcePrinter
+	if o.Preview {
+		printer, err = o.previewPrintFlags.ToPrinter(o.PreviewFormat)
+		if err != nil {
+			return err
+		}
+	}
+
+	info, err := fuzzyfinder.Infos(infos, printer, false, o.RawPreview)
```

Comment:
I thought with `resource.Info` I could use the existing fuzzyfinder implementation.

Reply:
cool! 👍 
I will fix the existing code.